codegen/header_propagate.go

```diff
@@ -70,6 +70,17 @@ func (hp *HeaderPropagator) Propagate(
 				val.QualifiedName, field.Name)
 		}
 
+		if val.StaticValue != "" {
```

Comment:
can we do type switch here to emphasis 2 different approach (static value vs from field)

Reply:
sure